src/client/Client.cc

```diff
@@ -14011,16 +14011,16 @@ bool Client::is_quota_bytes_exceeded(Inode *in, int64_t new_bytes,
 
 bool Client::is_quota_bytes_approaching(Inode *in, const UserPerm& perms)
 {
+  ceph_assert(in->size >= in->reported_size);
+  const uint64_t size = in->size - in->reported_size;
   return check_quota_condition(in, perms,
-      [](const Inode &in) {
+      [&size](const Inode &in) {
         if (in.quota.max_bytes) {
           if (in.rstat.rbytes >= in.quota.max_bytes) {
             return true;
           }
 
-          ceph_assert(in.size >= in.reported_size);
           const uint64_t space = in.quota.max_bytes - in.rstat.rbytes;
-          const uint64_t size = in.size - in.reported_size;
           return (space >> 4) < size;
```

Comment:
using size does not make sense when in != caller's in,

Reply:
yes

Reply:
At first we use this change in dir's used space to evaluate current dir's quota_bytes whether approaching, but dir not record this change in client side, and for each writing file we can get it.
     The claller's in is file and other test statement's in is dir, and the caller's in is a children of front dir. when the file size changing will lead dir used space changing, so i want to use the caller's 
file size change replace dir's used space change, but when multiple cephfs clients parallel writing files will reduce accuracy.

Reply:
@ukernel Ping?